mathesar/models/base.py

```diff
@@ -283,6 +283,60 @@ class Explorations(BaseModel):
     description = models.CharField(null=True)
 
 
+class Form(BaseModel):
+    id = models.UUIDField(primary_key=True)
+    name = models.CharField()
+    version = models.IntegerField()
+    database = models.ForeignKey('Database', on_delete=models.CASCADE)
+    base_table_oid = models.PositiveBigIntegerField()
+    schema_oid = models.PositiveBigIntegerField()
+    description = models.CharField(null=True)
+    header = models.OneToOneField('FormHeader', on_delete=models.DO_NOTHING)
+    submission = models.ForeignKey('SubmissionSettings', on_delete=models.CASCADE)
+    is_public = models.BooleanField(default=False)
+
+
+class FormHeader(BaseModel):
+    title = models.CharField()
+    subtitle = models.CharField(null=True)
+
+
+class SubmissionSettings(BaseModel):
+    role = models.ForeignKey('ConfiguredRole', on_delete=models.CASCADE)
+    message = models.CharField(null=True)
+    redirect_url = models.URLField(null=True)
+    submit_label = models.CharField(null=True)
+
+
+class FormField(BaseModel):
+    id = models.CharField(primary_key=True)
```

Comment:
@pavish Why does this `id` need to be a string?

Reply:
In the future, we would like the user to send a REST/RPC request directly to fill a form. These ids will serve as the keys in the request being sent, so it's important to be able to customize them. However, I agree that this should not be string.

We could follow the same pattern we follow for form id here, it can be an `uuid`.

In the future, we would have a separate unique field to represent the request key (similar to how we handle slug in the Form model), and we'd just copy this value over as its default.

Short answer:
- This should be a Primary Key.
- This needs to be an UUID v4.